src/utils/Utils.ts

```diff
@@ -1,28 +1,35 @@
-import { ResourceType, ResourceTypeAttribute, ResourceInstance } from '@/apis/rembrandt/rembrandt';
-import { ListElement } from '@/components/ListSection.vue';
+import { ResourceType, ResourceTypeAttribute } from '@/apis/rembrandt/rembrandt';
+import { LiOptions } from '@/components/Li.vue';
+
+export type clickHandler = (id: string) => void;
 
 export default class Utils {
   // region public static methods
-  public static resourceTypesToList(resourceTypes: ResourceType[]): ListElement[] {
+  public static resourceTypesToList(resourceTypes: ResourceType[], onClick?: clickHandler): LiOptions[] {
     return resourceTypes.map((resourceType) => {
       return {
         id: resourceType.id || resourceType.name,
         firstValue: resourceType.name,
         secondValue: resourceType.parentType ? `Parent Type: ${resourceType.parentType}` : '',
         thirdValue: resourceType.abstract ? 'Abstract Type' : `300 Instances`,
-        link: `/types/${resourceType.id}`,
+        link: {
+          link: onClick ? '' : `/types/${resourceType.id}`,
+          onClick: onClick ? () => { onClick(resourceType.id || resourceType.name); } : undefined,
+        },
       };
     });
   }
 
-  public static resourceTypeAttributesToList(resourceTypeAttributes: ResourceTypeAttribute[]): ListElement[] {
-    return resourceTypeAttributes.map((resourceTypeAttribute) => {
+  public static resourceTypeAttributesToList(attributes: ResourceTypeAttribute[], onClick?: clickHandler): LiOptions[] {
+    return attributes.map((attribute) => {
```

Comment:
if not resource type attributes, what kind of attributes are it then? resourceTypeAttribute seems a bit more expressive to me.

Reply:
But it is not necessary here since the Type of the parameter is specified.